### PR TITLE
updated docstring for set_lr_mult and set_wd_mult

### PR DESCRIPTION
**Pull request by bikestra**
screen shot 2017-05-04 at 11 45 55 pm

screen shot 2017-05-04 at 11 45 45 pm

@mli @zackchase @madjam @nswamy

### CODE REVIEW

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -174,14 +174,24 @@ def set_lr_scale(self, args_lrscale): # pylint: disable=unused-argument
...
 
     def set_lr_mult(self, args_lr_mult):
-        """Set individual learning rate for each weight.
+        """Set an individual learning rate multiplier for each parameter.
+
+        If you specify a learning rate multiplier for a parameter, then
+        the learning rate for the parameter will be set as the product of
+        the global learning rate `self.lr` and its multiplier.
+
+        .. note:: The default learning rate multiplier of a `Symbol`
```
Comment (piiswrong) [inline python/mxnet/optimizer.py]: you can only set it for variables and its lr_mult (without the underscore) in the interface

File: python/mxnet/optimizer.py [hunk 2]
```diff
@@ -174,14 +174,24 @@ def set_lr_scale(self, args_lrscale): # pylint: disable=unused-argument
...
+            can be set with its ``__lr_mult__`` `attr`.
 
         Parameters
         ----------
         args_lr_mult : dict of string/int to float
-            Set the lr multipler for name/index to float.
-            Setting multipler by index is supported for backward compatibility,
-            but we recommend using name and symbol.
+            For each of its key-value entry,
```
Comment (piiswrong) [inline python/mxnet/optimizer.py]: explain separately.
If you use index, it should be the same as the index you use for the weight in update method.
if you use string name, you also need to pass sym (which represent the network) in contructor

File: python/mxnet/optimizer.py [hunk 3]
```diff
@@ -174,14 +174,24 @@ def set_lr_scale(self, args_lrscale): # pylint: disable=unused-argument
         raise DeprecationWarning
 
     def set_lr_mult(self, args_lr_mult):
-        """Set individual learning rate for each weight.
+        """Set an individual learning rate multiplier for each parameter.
```
Comment (zackchase) [inline python/mxnet/optimizer.py]: "Set" -> "Sets"

File: python/mxnet/optimizer.py (hunk 2 above)
Comment (zackchase) [inline python/mxnet/optimizer.py]: "For each of its key-value entry" -> "entries"

File: python/mxnet/optimizer.py [hunk 4]
```diff
@@ -192,17 +202,25 @@ def set_lr_mult(self, args_lr_mult):
         self.lr_mult.update(args_lr_mult)
 
     def set_wd_mult(self, args_wd_mult):
-        """Set individual weight decay for each weight.
+        """Set an individual weight decay multiplier for each parameter.
```
Comment (zackchase) [inline python/mxnet/optimizer.py]: "Set" -> "Sets"

File: python/mxnet/optimizer.py [hunk 5]
```diff
@@ -192,17 +202,25 @@ def set_lr_mult(self, args_lr_mult):
...
 
-        By default wd multipler is 0 for all params whose name doesn't
-        end with _weight, if param_idx2name is provided.
+        By default, if `param_idx2name` was provided in the
+        constructor, weight decay multipler is set as 0 for all
```
Comment (zackchase) [inline python/mxnet/optimizer.py]: "weight decay multiplier" -> "the weight decay multiplier"

File: python/mxnet/optimizer.py [hunk 6]
```diff
@@ -192,17 +202,25 @@ def set_lr_mult(self, args_lr_mult):
...
+        .. note:: The default weight decay multiplier for a `Symbol`
+            can be set with its ``__wd_mult__`` `attr`.
 
         Parameters
         ----------
         args_wd_mult : dict of string/int to float
-            Set the wd multipler for name/index to float.
-            Setting multipler by index is supported for backward compatibility,
-            but we recommend using name and symbol.
+            For each of its key-value entry,
```
Comment (zackchase) [inline python/mxnet/optimizer.py]: "entry" -> "entries"

File: python/mxnet/optimizer.py (hunk 1 above)
Comment (bikestra) [inline python/mxnet/optimizer.py]: https://github.com/dmlc/mxnet/pull/6120/commits/7a769fa49ca540fc202056d7aab1eb156fdb2d88#diff-c445cb799a50a6e48c48d19a1ef37d3bL191 uses ``__lr_mult__`` attribute, so that's what I wanted to mention here. Do you think my description is incorrect?

Comment (piiswrong) [inline python/mxnet/optimizer.py]: yeah but its hidden from users. Users should specify this with mx.sym.Variable(..., lr_mult=0.1)

Comment (piiswrong) [inline python/mxnet/optimizer.py]: @bikestra  could you change this to 
`The learning rate multiplier of a Variable can be set with its lr_mutl argument on construction`
so we can merge.

**Comment by nswamy** [comment]
@piiswrong looks like all the comments are addressed on this one. can you merge this?
